source/common/redis/command_splitter_impl.cc

```diff
@@ -325,7 +325,25 @@ InstanceImpl::InstanceImpl(ConnPool::InstancePtr&& conn_pool, Stats::Scope& scop
 }
 
 SplitRequestPtr InstanceImpl::makeRequest(const RespValue& request, SplitCallbacks& callbacks) {
-  if (request.type() != RespType::Array || request.asArray().size() < 2) {
+  if (request.type() != RespType::Array) {
```

Comment:
Would it make more sense to do this at the proxy filter layer vs. the upstream command splitter? I could go either way but just throwing it out there.

Reply:
i don't see a compelling reason to start pushing what is effectively validation logic up to the proxy_filter. right now the proxy_filter calls decode() and throws it over the wall.

this code does need a cleanup/refactor at some point but i don't think it's worth starting in this PR. probably when we start doing replication work. happy to make changes if you end up wanting them though.

Reply:
OK fair enough. I will review tomorrow.